Wordle.Api/Wordle.Api/Services/PlayerServices.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Wordle.Api.Dtos;
+using Wordle.Api.Models;
+
+namespace Wordle.Api.Services
+{
+    public class PlayerServices(WordleDbContext Db)
+    {
+        private static object _lock = new();
+        public WordleDbContext Db { get; set; } = Db;
+
+
+        public async Task SavePlayer(Score score)
+        {
+            Player? player = await Db.Players.FirstOrDefaultAsync(player => player.Name == score.Name);
+
+            if (player is null) {
+                lock (_lock)
+                {
+                    player = new()
+                    {
+                        Name = score.Name,
+                        GameCount = 1,
+                        AverageAttempts = score.Attempts,
+                        AverageSeconds = score.Seconds
+                    };
+
+                    Db.Players.Add(player);
+                }
+
+            }
+            else
+            {
+                player.GameCount++;
+                player.AverageAttempts = (score.Attempts + player.AverageAttempts) / (player.GameCount);
+                player.AverageSeconds = (score.Seconds + player.AverageSeconds) / (player.GameCount);
+            }
+            Db.SaveChanges();
```

Comment:
```suggestion
            await Db.SaveChangesAsync();
```
When we're in an asynchronous method, we can take advantage of any methods available to us that have an async counterpart and await them. This way we don't block the thread and free it up to perform other tasks.

Reply:
thank you for the suggestion!